gateway/src/lib.rs

```diff
@@ -880,6 +881,74 @@ impl Actor {
                         VoteExecutionStatus::ConsensusReached => {}
                     }
 
+                    // we reach consensus in the checkpoints submission
+                    st.last_cron_executed_epoch = params_epoch;
+
+                    let msgs = submission
+                        .load_most_submitted_checkpoint(store)?
+                        .unwrap()
+                        .top_down_msgs;
+                    hamt.delete(&epoch_key)?;
+
+                    Ok(Some(msgs))
+                })
+                .map_err(|e| {
+                    log::error!(
+                        "encountered error processing submit cron checkpoint: {:?}",
+                        e
+                    );
+                    actor_error!(unhandled_message, e.to_string())
+                })
+        })?;
+
+        if let Some(msgs) = msgs {
+            for m in msgs {
+                Self::apply_msg_inner(
+                    rt,
+                    CrossMsg {
+                        msg: m,
+                        wrapped: false,
+                    },
+                )?;
+            }
+        }
+        Ok(RawBytes::default())
+    }
+
+    /// Externally trigger cron submission epoch. This is an edge case to ensure none of the epoches
```

Comment:
I don't think we need this external function, we can include a lazy check with every cron checkpoint submission to see if there is something to be executed. We could even check when something is executed if any subsequent epoch should also be executed. The issue I see with the latter approach is that if you have more than execution pending this may run our of gas, that is why I think is better to run just one execution per submission. 

The logic could be integrated in submit_cron and work as follows:
- The submission is performed. 
- If the submission triggers an execute we do so and it is done. 
- If not, we check the last executed and the next epoch and see if it can be executed. This lazy check will ensure that no checkpoint is ever stuck.